### PR TITLE
Implement custom moe op for Mixtral

### PR DESCRIPTION
**Pull request by jbyczkow**
Enable new mixture_of_experts op for bf16 route. When using quantization, run with legacy moe solution.

### CODE REVIEW

**Review by szutenberg — CHANGES_REQUESTED** [review]
Hi @jbyczkow,

I ran three tests on the latest 1.18.0 docker, your commit 9b99f7f48 and commit 48215c11e9 in our internal benchmarking repo:

- two tests finished with bad accuracy,
- one test crashed with Internal Server Error

Please double check your change.

File: vllm/hpu/ops.py [hunk 1]
```diff
@@ -253,41 +253,64 @@ def forward(self, state):
 
 class StaticFusedMOE(torch.nn.Module):
 
-    def __init__(self, num_total_experts):
+    def __init__(self, num_total_experts, using_inc_quantization):
```
Comment (MrGeva) [inline vllm/hpu/ops.py:256]: its not really controlling wether to use inc or not. this is just an artifact. its controlling dynamic vs. static moe, so please rename.

Comment (jbyczkow) [reply vllm/hpu/ops.py]: I updated to use_static_implementation. I also changed StaticFusedMOE class name to FusedMOE since we have static and dynamic paths now.

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 2]
```diff
@@ -258,21 +264,21 @@ def weight_loader(self, param: torch.nn.Parameter,
             if shard_id == 0:
                 param_data[expert_id,
                            0:shard_size, :] = loaded_weight[shard, :]
-                if is_hpu():
-                    self.hpu_static_fused_moe.w13_list[expert_id].set_weight(
+                if is_hpu() and hasattr(self.hpu_fused_moe, 'w13_list'):
```
Comment (dudilester) [inline vllm/model_executor/layers/fused_moe/layer.py:267]: Maybe to make the condition more clear instead of hasattr we can check that isinstance(self.hpu_fused_moe, StaticFusedMOE)

**Review by MrGeva — APPROVED** [review]

**Review by szutenberg — APPROVED** [review]
